iree/turbine/kernel/wave/codegen.py

```diff
@@ -272,6 +272,14 @@ def addi(lhs, rhs):
         if get_const_val(rhs) == 0:
             return lhs
 
+        if (rhs_val := get_const_val(rhs)) and rhs_val < 0:
+            # If rhs_val is negative, since it is of index type and we have
+            # overflow flags, we need to express it as a sub to get the
```

Comment:
Can you explain a little more why we'd need the overflow_flags? in theory, wouldn't be rhs_val  always be <= lhs?

Reply:
So basically what happens is that we are trying to do 
```
x - 2
```

Now this lowers to
```
%c_2 = arith.constant -2 : index
addi %x, %c_2, nsw, nuw
``` 
because sympy represents these are additions.
So the problem here is that we end up -2 with index type which is not really -2. What we really want is
```
%c2 = arith.constant 2 : index
subi %x, %c2 
``` 

Now the interesting this is that if we get rid of nsw, nuw, then the addi example works because we allow the overflow to happen. But we want to avoid overflow and so this was a solution to it, but happy to discuss other better solutions.

Reply:
I see I see, fwiw, I don't think we'd need a full discussion on this, just more for knowledge and maybe NIT haha.

but looks like your current change should generate:
```
%c2 = arith.constant 2 : index
subi %x, %c2 
```
with the `arith_d.subi` right? Dosn't that mean we won't need the "overflow" flag anymore?

Reply:
That's an interesting example (BTW, we can also check this issue on alive2 https://alive2.llvm.org/ce/z/dpybHs). There may be still similar issues lurking around, but I'd prefer to keep overflow flags.